crates/loader/src/assets.rs

```diff
@@ -90,6 +90,15 @@ pub fn file_sha256_digest_string(path: impl AsRef<Path>) -> std::io::Result<Stri
     Ok(digest_string)
 }
 
+/// Return the hex-encoded SHA256 digest of the given bytes.
+pub fn sha256_digest_string(data: &[u8]) -> String {
```

Comment:
This could be used in this file above at line 80 (and a few other places if the dependencies made sense).

At what point do we add a `spin-util` crate? :grimacing:

Reply:
This is funny coincidence, but I'm currently trying to extract similar stuff from my draft [PR](https://github.com/fermyon/spin/pull/884).

https://github.com/etehtsea/spin/commit/d9e0a5498e4e82ebdea930dcf3c15b880526d9c5

My two cents about spin-util:
there is an opinion that util-like libraries is a sort of "anti-pattern" where all the code "I don't know where to put it" is living, but I don't have any better idea at the moment.

Other candidates might be:
- `app_dir` which gets parent of spin.toml and fails with an error or panics when there is none.
- `absolutize`/`canonicalize` (wrappers around std/path-absolutize/dunce)